### ISSUE TITLE
Doesn't work in IE11

### ISSUE THREAD

**Issue by altano** [jakearchibald/es6-promise#109]
It looks like MutationObserver isn't firing in IE11.  Paste this into the console:

```javascript
ES6Promise.Promise.resolve("hello").then(function(v){console.log(v);});
```
This won't fire.  If I then paste this:
```javascript
setTimeout(function(){},1);
```
Then both the setTimeout event AND the `.then()` callback function get called.  This must be some kind of IE bug?  I couldn't find anyone reporting it though so I must be mistaken?

**Comment by altano** [comment]
Also worth noting: $$asap$$flush WILL get called immediately if I set a breakpoint in es6-promise on the line that changes node.data.  If I disable the breakpoint, $$asap$$flush doesn't get called.  Bizarre, right?

**Comment by stefanpenner** [comment]
Woah. Maybe a mutation observer related quirk? This seems gnarly. I'll take a look in the AM

**Comment by edoloughlin** [comment]
Any update on this?

**Comment by stefanpenner** [comment]
Sorry this slipped my queue I'll take a look today

**Comment by stefanpenner** [comment]
I am unfortunately unable to reproduce this issue. :( 
On IE11 all but 1 test is passing, and that test failure appears totally unrelated.

**Comment by stefanpenner** [comment]
So, this is interesting. When stepping through the code, everything works as expected :S

**Comment by stefanpenner** [comment]
Also, if you merely "click" anywhere in the page, the queue flushes..

**Comment by stefanpenner** [comment]
MessageChannel appears to work correctly...

**Comment by stefanpenner** [comment]
I suspect the resolution is to:

1. switch to MessageChannel
2. report the bug upstream.

**Comment by lun471k** [comment]
Any update on the issue? You're saying it's related to the console... do you mean simply having the console/dev tools opened, or using the `console` javascript object (`console.log()`) ? 

I'm facing a similar issue with the promise not resolving correctly on IE11 and I'm wondering if this is the problem.

**Comment by stefanpenner** [comment]
This is only the result of triggering a change from the console. So when a dev attempts to debug a promise via the developer console. They must tap the web view to flush the changes. This is clearly and ie11 bug. Although to save the sanity of dev, we may want to migrate to message channel for async.

If the change did not originate from the console, I have very high confidence that it is not a bug with this library. As its extensive test suite runs without issue on ie11.

To summarize, this does not affect normal app usage. Only some circumstances in the ie11 dev console do issues arise.

**Comment by stefanpenner** [comment]
So we have a choice to make, we can revert to use the wrong async in IE11, or deal with it. Both options really do suck..

cc @jdalton I tried to report this but with IE11, but the login sequence kept erroring with me. Do you have any thoughts or suggestions here?

cc @domenic Also curious if you have some advice. Let me know if i should provide more context.

**Comment by domenic** [comment]
I would say deal with it; devs should be deving for Edge hopefully. Maybe add a loud warning.

**Comment by stefanpenner** [comment]
> Maybe add a loud warning.

sniff the version and console.log something ?

**Comment by domenic** [comment]
Yeah or just in the readme. But that is probably better.

**Comment by lun471k** [comment]
Either solution would probably fit everyone's needs. IE11 will be forgotten soon enough with Edge being pushed out so much.

**Comment by stefanpenner** [comment]
I can likely also give a function to force the lib to some other async. Something the dev can run right after seeing the message so debugging isn't horrible

**Comment by jdalton** [comment]
fwiw I can't repro the issue on Win 10 IE 11 using http://static.iamstef.net/es6-promise/ or `ES6Promise.Promise.resolve("hello").then(function(v){console.log(v);});`.

**Comment by jdalton** [comment]
Yall running updated OS's and what not?

**Comment by jdalton** [comment]
Confirmed it worked on my Win 10 vm and https://remote.modern.ie/ too.

**Comment by jdalton** [comment]
Works as in cannot repro bug.

**Comment by stefanpenner** [comment]
> Works as in cannot repro bug.

If this is accurate, then awesome. Maybe some patch addressed this.

I am going to close the issue, as @jdalton someone who works on IE/chakra is unable to reproduce. I would say the problem is fixed.

If someone can demonstrate another scenario that fails we will gladly reopen.

@jdalton / @domenic thanks for your help
